src/test/unit/pid_unittest.cc

```diff
@@ -618,51 +596,51 @@ TEST(pidControllerTest, testCrashRecoveryMode)
-    setStickPosition(FD_PITCH, -1.0f);
-    setStickPosition(FD_YAW, -1.0f);
-
-    pidController(pidProfile, currentTestTime());
-
-    EXPECT_NEAR(2232.78, pidData[FD_ROLL].F, calculateTolerance(2232.78));
-    EXPECT_NEAR(-2061.03, pidData[FD_PITCH].F, calculateTolerance(-2061.03));
-    EXPECT_NEAR(-2061.03, pidData[FD_YAW].F, calculateTolerance(-2061.03));
-
-    // Match the stick to gyro to stop error
-    setStickPosition(FD_ROLL, 0.5f);
-    setStickPosition(FD_PITCH, -0.5f);
-    setStickPosition(FD_YAW, -0.5f);
-
-    pidController(pidProfile, currentTestTime());
-
-    EXPECT_NEAR(-558.20, pidData[FD_ROLL].F, calculateTolerance(-558.20));
-    EXPECT_NEAR(515.26, pidData[FD_PITCH].F, calculateTolerance(515.26));
-    EXPECT_NEAR(515.26, pidData[FD_YAW].F, calculateTolerance(515.26));
-
-    setStickPosition(FD_ROLL, 0.0f);
-    setStickPosition(FD_PITCH, 0.0f);
-    setStickPosition(FD_YAW, 0.0f);
-
-    for (int loop = 0; loop <= 15; loop++) {
-        gyro.gyroADCf[FD_ROLL] += gyro.gyroADCf[FD_ROLL];
-        pidController(pidProfile, currentTestTime());
-    }
-
-    EXPECT_FLOAT_EQ(0, pidData[FD_ROLL].F);
-    EXPECT_FLOAT_EQ(0, pidData[FD_PITCH].F);
-    EXPECT_FLOAT_EQ(0, pidData[FD_YAW].F);
-}
+// TEST(pidControllerTest, testFeedForward) // now in rc.c
+// {
+//     resetTest();
+//     ENABLE_ARMING_FLAG(ARMED);
+//     pidStabilisationState(PID_STABILISATION_ON);
+// 
+//     EXPECT_FLOAT_EQ(0, pidData[FD_ROLL].F);
+//     EXPECT_FLOAT_EQ(0, pidData[FD_PITCH].F);
+//     EXPECT_FLOAT_EQ(0, pidData[FD_YAW].F);
+// 
+//     // Match the stick to gyro to stop error
+//     setStickPosition(FD_ROLL, 1.0f);
+//     setStickPosition(FD_PITCH, -1.0f);
+//     setStickPosition(FD_YAW, -1.0f);
+// 
+//     pidController(pidProfile, currentTestTime());
+// 
+//     EXPECT_NEAR(2232.78, pidData[FD_ROLL].F, calculateTolerance(2232.78));
+//     EXPECT_NEAR(-2061.03, pidData[FD_PITCH].F, calculateTolerance(-2061.03));
+//     EXPECT_NEAR(-2061.03, pidData[FD_YAW].F, calculateTolerance(-2061.03));
+// 
+//     // Match the stick to gyro to stop error
+//     setStickPosition(FD_ROLL, 0.5f);
+//     setStickPosition(FD_PITCH, -0.5f);
+//     setStickPosition(FD_YAW, -0.5f);
+// 
+//     pidController(pidProfile, currentTestTime());
+// 
+//     EXPECT_NEAR(-558.20, pidData[FD_ROLL].F, calculateTolerance(-558.20));
+//     EXPECT_NEAR(515.26, pidData[FD_PITCH].F, calculateTolerance(515.26));
+//     EXPECT_NEAR(515.26, pidData[FD_YAW].F, calculateTolerance(515.26));
+// 
+//     setStickPosition(FD_ROLL, 0.0f);
+//     setStickPosition(FD_PITCH, 0.0f);
+//     setStickPosition(FD_YAW, 0.0f);
+// 
+//     for (int loop = 0; loop <= 15; loop++) {
+//         gyro.gyroADCf[FD_ROLL] += gyro.gyroADCf[FD_ROLL];
+//         pidController(pidProfile, currentTestTime());
+//     }
+// 
+//     EXPECT_FLOAT_EQ(0, pidData[FD_ROLL].F);
+//     EXPECT_FLOAT_EQ(0, pidData[FD_PITCH].F);
+//     EXPECT_FLOAT_EQ(0, pidData[FD_YAW].F);
+// }
```

Comment:
Do we still need this?
```
// TEST(pidControllerTest, testFeedForward) // now in rc.c
// {
//     resetTest();
//     ENABLE_ARMING_FLAG(ARMED);
//     pidStabilisationState(PID_STABILISATION_ON);
// 
//     EXPECT_FLOAT_EQ(0, pidData[FD_ROLL].F);
//     EXPECT_FLOAT_EQ(0, pidData[FD_PITCH].F);
//     EXPECT_FLOAT_EQ(0, pidData[FD_YAW].F);
// 
//     // Match the stick to gyro to stop error
//     setStickPosition(FD_ROLL, 1.0f);
//     setStickPosition(FD_PITCH, -1.0f);
//     setStickPosition(FD_YAW, -1.0f);
// 
//     pidController(pidProfile, currentTestTime());
// 
//     EXPECT_NEAR(2232.78, pidData[FD_ROLL].F, calculateTolerance(2232.78));
//     EXPECT_NEAR(-2061.03, pidData[FD_PITCH].F, calculateTolerance(-2061.03));
//     EXPECT_NEAR(-2061.03, pidData[FD_YAW].F, calculateTolerance(-2061.03));
// 
//     // Match the stick to gyro to stop error
//     setStickPosition(FD_ROLL, 0.5f);
//     setStickPosition(FD_PITCH, -0.5f);
//     setStickPosition(FD_YAW, -0.5f);
// 
//     pidController(pidProfile, currentTestTime());
// 
//     EXPECT_NEAR(-558.20, pidData[FD_ROLL].F, calculateTolerance(-558.20));
//     EXPECT_NEAR(515.26, pidData[FD_PITCH].F, calculateTolerance(515.26));
//     EXPECT_NEAR(515.26, pidData[FD_YAW].F, calculateTolerance(515.26));
// 
//     setStickPosition(FD_ROLL, 0.0f);
//     setStickPosition(FD_PITCH, 0.0f);
//     setStickPosition(FD_YAW, 0.0f);
// 
//     for (int loop = 0; loop <= 15; loop++) {
//         gyro.gyroADCf[FD_ROLL] += gyro.gyroADCf[FD_ROLL];
//         pidController(pidProfile, currentTestTime());
//     }
// 
//     EXPECT_FLOAT_EQ(0, pidData[FD_ROLL].F);
//     EXPECT_FLOAT_EQ(0, pidData[FD_PITCH].F);
//     EXPECT_FLOAT_EQ(0, pidData[FD_YAW].F);
// }
```

Reply:
Thanks, Mark.  I left it there only because I have not been able to successfully port these unit tests across, now the feedforward code is in rc.c.  I simply cannot figure how to do it.  So these are left here as a 'reminder' of how the old tests were constructed.  Once we have the new unit tests in place, this can go.

Reply:
Why are you removing tests? Those are exactly there to proove the flight behaviour is unchanged with refactoring.

Reply:
Agree we need to rewrite the tests before merging

Reply:
I remove them from this place, because they cannot work in this location. 
I don't have a sufficient understanding of unit tests to implement the same tests, given that now that the code been moved new location.
I've asked many times for help with the unit tests.

Reply:
I checked these tests, they are somewhat useless.  They calculate feedforward from simulated setpointDelta values, and apply a transition value, but neither of those bits of maths actually check that the feedforward value is properly derived from changes in setpoint.

I've updated the pid_unittest so that it performs basically as before, in that it makes and artificial setpoint from stick values, computes a primitive setpointDelta, and then applies feedforward gain.  

We will need to create a proper feedforward unittest.  It probably will need to be in rc.c, where both the setpoint and feedforward code exists.  I've started work on that on [a local branch of mine](https://github.com/betaflight/betaflight/commit/5e94aeffac6c56ea21fd90ef4cddc24b9927d618)...

Currently the unit test on that branch fails in this manner, and I can't fix it.  If someone knows how, please share:-) :

```
compiling unit/rc_unittest.cc 
unit/rc_unittest.cc:157:5: error: use of undeclared identifier 'calculateFeedforward'
    calculateFeedforward(&pidRuntime, FD_ROLL); // **** FAILS !! *******
```

Reply:
@ctzsnooze well not useless as you want to do tests as much as possible in isolation without too many external factors tested somewehere else. 
I haven't really looked too much in your code yet to understand it properly in how it can be fixed

Reply:
Thanks @borisbstyle - I have fixed the tests that can be done in the pid_unittest now.
Am working on proper tests of feedforward maths itself, in the rc.c side of things.
These were never done, previously.